#!/usr/bin/env python

import sys
import json
sys.path.insert(0, "/opt/handler/lib/python")
from eupath import BiomFileMicrobiomeDbExporter
from handler_base.dataset_handler import ValidationException
import optparse


def main():
    """
      The following program is a Galaxy Tool for exporting a BIOM file from Galaxy to EuPathDB via iRODS.

      Sample for testing outside of Galaxy (assuming the existing file structure):
      python exportGeneListToEuPathDB
             "a name" "a summary" "a description" "<user.wdk_id@eupath.org eg. crisl.108976930@eupathdb.org>"
             "<the galaxy tool xml directory e.g. ../lib/xml >" "output" ""
             "<path to biom file>"
    """

    parser = optparse.OptionParser()
    (options, args) = parser.parse_args()

    # So users are greeted with a bewildering traceback
    # copied from gene list exporter
    sys.tracebacklimit = 0

    exporter = BiomFileMicrobiomeDbExporter.BiomExport(args)
    try:
        exporter.export()
    except ValidationException as ve:
        print >> sys.stderr, json.dumps({'error': 'user', 'message': str(ve)})
        sys.exit(1)
    except BaseException as err:
        print >> sys.stderr, json.dumps({'error': 'unknown', 'message': str(err)})
        sys.exit(1)


if __name__ == "__main__":
    sys.exit(main())
